### PR TITLE
What's On: `ExhibitionPage`

### PR DESCRIPTION
**Pull request by jamesbiggs**
## About these changes

Updates the existing (alpha) Exhibition page model to meet the new spec.

Also covers [DSRC-59] + [DSRC-131] in this PR.

Changed `type_label` to be a `cached_property` rather than a `classmethod` so that we can change the `type_label` based on the instance rather than the class itself (necessary to make an Exhibition a "Past exhibition" when today's date is past the end date) - this affected the website search slightly but this is due for removal as it'll be superseded by the Search project.

Also added some text choices and mixins for brand and accent colours - potentially useful in the future for applying brand colours across the site.

[DSRC-59]: https://national-archives.atlassian.net/browse/DSRC-59?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DSRC-131]: https://national-archives.atlassian.net/browse/DSRC-131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: etna/core/models/mixins.py
Comment (ahosgood) [inline etna/core/models/mixins.py:207]: This should instead be a dropdown of:

- none
- black
- pink
- orange
- yellow
- green
- blue

File: etna/whatson/models.py
Comment (ahosgood) [inline etna/whatson/models.py:1033]: Not sure if we need this property. I can compute this in the frontend based on `price`.

File: etna/whatson/models.py
Comment (ahosgood) [inline etna/whatson/models.py:758]: Should the hero image be required for this page type? Do we need another mixin like `RequiredHeroImageMixin`?

Comment (jamesbiggs) [reply etna/whatson/models.py:758]: Missed the asterisk on the spec - fixing now!

File: etna/whatson/models.py
Comment (ahosgood) [inline etna/whatson/models.py:777]: Should this just be a DateField rather than DateTimeField?

File: etna/whatson/models.py
Comment (jamesbiggs) [reply etna/whatson/models.py:792]: The `booking_details` could be just "No booking needed" or it could be "Book on Eventbrite" with a link in

File: etna/whatson/models.py
Comment (ahosgood) [inline etna/whatson/models.py:787]: Will we ever need ticket prices like £4.50?

File: etna/whatson/models.py
Comment (ahosgood) [reply etna/whatson/models.py:799]: Could we disallow bold? Only having links as an option?

File: etna/core/models/mixins.py
Comment (ahosgood) [reply etna/core/models/mixins.py]: I'm assuming these are the values I've changed

File: etna/core/blocks/shop.py
Comment (ahosgood) [inline etna/core/blocks/shop.py:34]: Is this not required?

File: etna/core/blocks/quote.py
Comment (ahosgood) [inline etna/core/blocks/quote.py:28]: At the moment, this is coming out in the API as a string. Could it be an integer instead?

File: etna/core/models/mixins.py
Comment (ahosgood) [inline etna/core/models/mixins.py:231]:
```suggestion
    hero_style = models.CharField(
```

File: etna/core/models/mixins.py
Comment (ahosgood) [inline etna/core/models/mixins.py:243]:
```suggestion
        FieldPanel("hero_style"),
```

File: etna/core/models/mixins.py
Comment (ahosgood) [inline etna/core/models/mixins.py:247]:
```suggestion
        APIField("hero_style"),
```

File: etna/core/models/mixins.py
Comment (jamesbiggs) [reply etna/core/models/mixins.py:231]: Thanks, I didn't like this name 😆

File: etna/core/colours.py
Comment (ahosgood) [inline etna/core/colours.py:29]:
```suggestion
class HeroColourChoices(models.TextChoices):
    """
    This model is a list of our "hero" colour choices, which can be used
    on the hero component.
    """

    NONE = "none", _("None")
    CONTRAST = "contrast", _("Contrast")
    TINT = "tint", _("Tint")
    ACCENT = "accent", _("Accent")


class HeroLayoutChoices(models.TextChoices):
    """
    This model is a list of our "hero" layouts, which can be used
    on the hero component.
    """

    DEFAULT = "default", _("Default")
    SHIFT = "shift", _("Shifted")
    SPLIT = "split", _("Split")
```

File: etna/core/models/mixins.py
Comment (ahosgood) [inline etna/core/models/mixins.py:220]: Could we maybe move this to another tab, maybe called "Design" as it is a field that affects the page as a whole?

**Review by ahosgood — APPROVED** [review]
Approved once the hero layout options are added and the page accent colour dropdown is potentially moved.

File: etna/core/models/mixins.py
Comment (ahosgood) [inline etna/core/models/mixins.py:252]:
```suggestion
class HeroStyleMixin(models.Model):
```

Comment (ahosgood) [reply etna/core/models/mixins.py:252]: Sorry to be pedantic!

Comment (ahosgood) [reply etna/core/models/mixins.py:252]: Once this is done, I am happy for it to be merged in 🚀

**Review by ahosgood — APPROVED** [review]

**Pull request merged**
